File: A_Arrival_of_the_General.java
import java.util.*;

public class A_Arrival_of_the_General {
    public static int calculateSwaps(int n, int[] heights) {
        int maxIdx = 0, minIdx = 0;

        for (int i = 1; i < n; i++) {
            if (heights[i] > heights[maxIdx]) maxIdx = i;
            if (heights[i] <= heights[minIdx]) minIdx = i;
        }

        int swaps = maxIdx + (n - 1 - minIdx);
        if (maxIdx > minIdx) swaps--;

        return swaps;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] heights = new int[n];
        for (int i = 0; i < n; i++) {
            heights[i] = sc.nextInt();
        }

        System.out.println(calculateSwaps(n, heights));
        sc.close();
    }
}

File: A_Arrival_of_the_General.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int calculateSwaps(int n, vector<int> &heights) {
    int maxIdx = max_element(heights.begin(), heights.end()) - heights.begin();
    int minIdx = min_element(heights.rbegin(), heights.rend()) - heights.rbegin();
    minIdx = n - 1 - minIdx;
    // this line is for verify commit
    int swaps = maxIdx + (n - 1 - minIdx);
    if (maxIdx > minIdx) {
        swaps--;
    }
    return swaps;
}

int main() {
    int n;
    cin >> n;
    vector<int> heights(n);

    for (int i = 0; i < n; i++) {
        cin >> heights[i];
    }

    cout << calculateSwaps(n, heights) << endl;
    return 0;
}
